head/src/main/kotlin/io/qalipsis/core/head/inmemory/StandaloneCampaignReportStateKeeperImpl.kt

```diff
@@ -186,10 +186,10 @@ internal class StandaloneCampaignReportStateKeeperImpl(
     }
 
     @LogInputAndOutput
-    override suspend fun retrieveCampaignReport(tenant: String, campaignKey: CampaignKey): CampaignExecutionDetails {
+    override suspend fun retrieveCampaignReport(tenant: String, campaignKeys: List<CampaignKey>): Collection<CampaignExecutionDetails> {
         join()
-        return campaignStates[campaignKey]?.map { (_, runningScenarioCampaign) ->
-            runningScenarioCampaign.toReport(campaignKey)
-        }?.toCampaignExecutionDetails() ?: throw IllegalArgumentException("No report found for the expected campaign")
+        return campaignKeys.map { campaignStates[it]?.map { (_, runningScenarioCampaign) ->
+            runningScenarioCampaign.toReport(it)
+        }?.toCampaignExecutionDetails() ?: throw IllegalArgumentException("No report found for the expected campaign") }
```

Comment:
```suggestion
        }?.toCampaignExecutionDetails() }
```

Reply:
What to do when the campaignStates map doesn't contain one of the campaignKeys ? I think it's necessary to throw the exception.

Reply:
no exception, just ignore when the key does not exists and only return the values that exist. However, if none can be found, I would return an IllegalArgumentException